File: angularFrontend/src/app/komentar.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class KomentarService {

  constructor(private http:HttpClient) { }



  findByPost(id):Observable<any>{
    return this.http.get("http://localhost:8080/api/komentari/"+id);
  }

  edit(data):Observable<any>{
    return this.http.put("http://localhost:8080/api/komentari/"+data.id,data);

  }
  findOne(id):Observable<any>{
    return this.http.get("http://localhost:8080/api/komentari/komentar/"+id);
  }

  delete(id):Observable<any>{
    return this.http.delete("http://localhost:8080/api/komentari/"+id)
  }
  
  save(data):Observable<any>{
   
    return this.http.post("http://localhost:8080/api/komentari",data);
  }
}

File: src/main/java/projekat/tiac/service/KomentarService.java
package projekat.tiac.service;

import java.util.List;
import java.util.Optional;

import projekat.tiac.model.Komentar;
import projekat.tiac.model.Post;

public interface KomentarService {
	
	public Komentar save (Long idPost, Komentar komentar);
	public Optional<Komentar> findOne(Long id);
	public List<Komentar> findByPost(Long idPost);
	public Komentar delete(Long id);
}
